src/VisualStudio/TestUtilities/IntegrationHelper.cs

```diff
@@ -0,0 +1,241 @@
+        }
+
+        public static IEnumerable<IntPtr> GetTopLevelWindows()
+        {
+            var topLevelWindows = new List<IntPtr>();
+
+            var enumFunc = new User32.WNDENUMPROC((hWnd, lParam) => {
+                topLevelWindows.Add(hWnd);
+                return true;
+            });
+            User32.EnumWindows(enumFunc, IntPtr.Zero);
+
+            return topLevelWindows;
+        }
+
+        public static void KillProcess(Process process)
+        {
+            if ((process != null) && (!process.HasExited))
+            {
+                process.Kill();
+            }
+        }
+
+        public static void KillProcess(string processName)
+        {
+            foreach (var process in Process.GetProcessesByName(processName))
+            {
+                KillProcess(processName);
+            }
+        }
+
+        public static DTE LocateDteForProcess(Process process)
+        {
+            object dte = null;
+            IRunningObjectTable runningObjectTable = null;
+            IEnumMoniker enumMoniker = null;
+            IBindCtx bindContext = null;
+            var monikers = new IMoniker[1];
+            var vsProgId = IntegrationHost.VsProgId;
+
+            Ole32.GetRunningObjectTable(0, out runningObjectTable);
+            runningObjectTable.EnumRunning(out enumMoniker);
+            Ole32.CreateBindCtx(0, out bindContext);
+
+            do
+            {
+                monikers[0] = null;
+
+                var monikersFetched = 0u;
+                var result = enumMoniker.Next(1, monikers, out monikersFetched);
+
+                if (result != 0)
+                {
+                    return null;
+                }
+
+                var moniker = monikers[0];
+                var fullDisplayName = null as string;
+
+                moniker.GetDisplayName(bindContext, null, out fullDisplayName);
+
+                var displayNameProcessId = 0;
+
+                if (!int.TryParse(fullDisplayName.Split(':').Last(), out displayNameProcessId))
+                {
+                    continue;
+                }
+
+                var displayName = fullDisplayName.Substring(0, (fullDisplayName.Length - (displayNameProcessId.ToString().Length + 1)));
+                var fullProgId = vsProgId.StartsWith("!") ? vsProgId : $"!{vsProgId}";
+
+                if (displayName.Equals(fullProgId, StringComparison.OrdinalIgnoreCase) &&
+                    (displayNameProcessId == process.Id))
+                {
+                    runningObjectTable.GetObject(moniker, out dte);
+                }
+            }
+            while (dte == null);
+
+            return (DTE)(dte);
+        }
+
+        public static async Task<string> LocateFile(string fileName, string downloadUrl = "")
+        {
+            if (File.Exists(fileName))
+            {
+                return Path.GetFullPath(fileName);
+            }
+
+            var filePath = Path.Combine(Environment.CurrentDirectory, fileName);
```

Comment:
Can we not depend on Environment.CurrentDirectory here? It's a recipe for all sorts of flakiness and bugs. I'm frankly not sure why we need a scenario where you can have a file name and need to probe a bunch of different places.